src/metrics.jl

```diff
@@ -308,6 +308,53 @@ Base.@propagate_inbounds function _evaluate(d::UnionMetrics, a::AbstractArray, b
     end
 end
 
+eval_op_a(d, ai, b) = eval_op(d, ai, zero(eltype(b)))
+eval_op_b(d, bi, a) = eval_op(d, zero(eltype(a)), bi)
+
+Base.@propagate_inbounds function _evaluate(d::UnionMetrics, a::SparseVector, b::SparseVector, ::Nothing)
```

Comment:
I wonder if you can relax this to some sort of `AbstractSparseVector`, which should ideally include column views into sparse matrices. Then `colwise` and `pairwise` should also benefit from this "out-of-the-box"™️.

Reply:
Wow. I already pushed something along these lines before reading this comment. It's not clear to me what interface `AbstractSparseVector` supports. But, `SparseColumnView` was easily supported. This is the vector you get from `@view m[:, i]` for a sparse matrix `m`.

Reply:
Awesome, the performance benefit is really significant. Your example is 100x faster for me! I think we will want to release this directly once it's merged, so you could bump the patch version in the Project.toml right away.